packages/nimble/inst/tests/test_utils.R

```diff
@@ -26,6 +26,49 @@ temporarilyAssignInGlobalEnv <- function(value) {
     do.call('on.exit', list(rmCommand), envir = parent.frame())
 }
 
+withTempProject <- function(code) {
+    code <- substitute(code)
+    project <- nimble:::nimbleProjectClass()
+    nimbleOptions(nimbleProject = project)
+    on.exit({
+        ## messages are suppressed by test_that, so "assign('.check', 1, globalenv())" can be used as a way to verify this code was called 
+        .project <- nimbleOptions()$nimbleProject
+        nimbleOptions(nimbleProject = NULL) ## clear this before clearCompiled step, in case clearCompiled() itself fails
+        .project$clearCompiled()
+    })
+    try(eval(code, envir = parent.frame()))
```

Comment:
On the second point: I now have verification of correct test failing in `test-testTools.R`